### PR TITLE
Fix broken TextArea autosizing and focus

### PR DESCRIPTION
**Pull request by Pedro-Everett**
**Backwards Compatibility Implications** 

_None_

**New Features** 

_None_

**Bug Fixes** 

- Fixed `TextArea` loosing its autosize and focus when rendering tooltips

### CODE REVIEW

File: src/components/Fields/TextArea.js [hunk 1]
```diff
@@ -98,6 +98,29 @@ class TextArea extends Component {
...
+        const isShowingErrorTooltip = !isValid && !!tooltipError;
+        const wasShowingErrorTooltip = !prevProps.isValid && !!prevProps.tooltipError;
+        if (
+            (hasFocus && isShowingErrorTooltip && !wasShowingErrorTooltip) ||
+            (hasFocus && !isShowingErrorTooltip && wasShowingErrorTooltip)
+        ) {
+            this.fixInputFocus();
+        }
+        autosize.destroy(this.inputRef);
+        autosize(this.inputRef);
```
Comment (jfrej) [inline src/components/Fields/TextArea.js]: We don't want to reset `autosize` on every component update. It's only needed when the tooltip is shown or hidden - so within the if block above. And I'd probably call `fixInputFocus` last to ensure the focus is not affected by the `autosize` reset.

File: src/components/Fields/TextArea.story.js [hunk 2]
```diff
@@ -11,6 +11,44 @@ Velit ipsam, nesciunt doloribus, aspernatur a itaque assumenda odit alias rem di
 
 Velit ipsam, nesciunt doloribus, aspernatur a itaque assumenda odit alias rem dignissimos culpa error magnam veniam cupiditate repellendus nam id perferendis sit.`;
 
+const MAX_COMMENT_LENGTH = 50;
+
+class CharartersLimitDemo extends Component {
```
Comment (jfrej) [inline src/components/Fields/TextArea.story.js]: Typo: `CharartersLimitDemo`/`CharactersLimitDemo`

File: src/components/Fields/TextArea.test.js [hunk 3]
```diff
@@ -304,4 +304,46 @@ describe('TextArea', () => {
...
+        textField.setState({ hasFocus: true });
+        sinon.spy(textField.instance(), 'fixInputFocus');
+
+        textField.setProps({
+            isValid: true,
+        });
+        expect(textField.instance().fixInputFocus).to.be.called();
+    });
+
+    it('fixInputFocs calls input ref focus', () => {
```
Comment (jfrej) [inline src/components/Fields/TextArea.test.js]: Typo:
```suggestion
    it('fixInputFocus calls input ref focus', () => {
```

Could you also fix this in the `TextField` tests?

File: src/components/Fields/TextArea.test.js [hunk 4]
```diff
@@ -304,4 +304,46 @@ describe('TextArea', () => {
...
+        const textField = mount(<TextArea />);
+        sinon.spy(textField.instance().inputRef, 'focus');
+        textField.instance().fixInputFocus();
+
+        expect(textField.instance().inputRef.focus).to.be.called();
+    });
+
+    it('fixInputFocs sets input the selection to the last character', () => {
```
Comment (jfrej) [inline src/components/Fields/TextArea.test.js]: Typo:
```suggestion
    it('fixInputFocus sets the input selection to the last character', () => {
```

File: src/components/Fields/TextArea.js (hunk 1 above)
Comment (Pedro-Everett) [reply src/components/Fields/TextArea.js]: @jfrej  Changed as suggested

File: src/components/Fields/TextArea.story.js (hunk 2 above)
Comment (Pedro-Everett) [reply src/components/Fields/TextArea.story.js]: Changed

File: src/components/Fields/TextArea.test.js (hunk 3 above)
Comment (Pedro-Everett) [reply src/components/Fields/TextArea.test.js]: Changed

File: src/components/Fields/TextArea.test.js (hunk 4 above)
Comment (Pedro-Everett) [reply src/components/Fields/TextArea.test.js]: Changed

File: src/components/Fields/TextArea.test.js [hunk 5]
```diff
@@ -304,4 +304,46 @@ describe('TextArea', () => {
...
+
+    it('fixInputFocus calls input ref focus', () => {
+        const textField = mount(<TextArea />);
+        sinon.spy(textField.instance().inputRef, 'focus');
+        textField.instance().fixInputFocus();
+
+        expect(textField.instance().inputRef.focus).to.be.called();
+    });
+
+    it('fixInputFocus sets input the selection to the last character', () => {
```
Comment (jfrej) [inline src/components/Fields/TextArea.test.js]: Looks like you missed my comment about the word "input" being in the wrong place:
```suggestion
    it('fixInputFocus sets the input selection to the last character', () => {
```

**Pull request merged**